inference/conflict.go

```diff
@@ -110,18 +110,18 @@ func pathString(nodes []node) string {
 
 // String converts a nilFlow to a string representation, where each entry is the flow of the form: `<pos>: <reason>`
 func (n *nilFlow) String() string {
+	var allNodes []node
+	allNodes = append(allNodes, n.nilPath...)
+	allNodes = append(allNodes, n.nonnilPath...)
+
 	flow := ""
-	for _, nodes := range [...][]node{n.nilPath, n.nonnilPath} {
-		for _, nodeObj := range nodes {
-			flow += nodeObj.String()
-		}
+	for _, nodeObj := range allNodes {
+		flow += nodeObj.String()
```

Comment:
Any reason for this change?

Reply:
Short answer is it was causing errors being dropped.

Long answer: Single assertion conflict has only one producer and consumer. Now with the concatenation logic, this got combined in a single node which is stored in the `nonnilPath`. Thus keeping `nilPath` nil. In such cases , `for _, nodes := range [...][]node{n.nilPath, n.nonnilPath} {` was causing the combined slice to be nil, causing the loop to be skipped, dropping errors.